devito/ir/equations/equation.py

```diff
@@ -188,12 +188,19 @@ def __new__(cls, *args, **kwargs):
         for d in ordering:
             if not d.is_Conditional:
                 continue
-            if d.condition is None:
+            if d._factor is not None:
```

Comment:
why not `d.factor`?

Reply:
`d.factor` is never None it returns `1` if there is no factor for various reasons